src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1042,7 +1042,7 @@ private static object DeepCopierHelper(Type t, object original)
         /// </summary>
         /// <param name="t">The type.</param>
         /// <returns>true if <paramref name="t"/> is serializable, false otherwise.</returns>
-        internal static bool HasSerializer(Type t)
+        public static bool HasSerializer(Type t)
```

Comment:
`Serialize` is also public ,and it indeed calls `SerializeInner`(also public) which in turn calls `GetSerializer` - which locks. I get your point though - we should not extend the surface of this. Those methods being public (also `Register`, etc.) is bad enough given stated concerns.

I will move everything to `TesterInternal`.